source/wp-content/themes/wporg-news-2021/sass/post/_header.scss

```diff
@@ -1,4 +1,240 @@
+			stroke: var(--bar-text-color);
+		}
+
+		.wp-site-blocks & {
+			margin-top: 0;
+		}
+
+		// Color variations
+		.category-development & {
+			--bar-background-image: url("assets/svg/brush-stroke-nav-bar-green.svg"),
+			url("data:image/svg+xml,%3Csvg height='1000' width='1' version='1.1' xmlns='http://www.w3.org/2000/svg'%3E%3Crect x='0' y='0' height='1000' width='1' r='0' rx='0' ry='0' fill='%2372d1a7'%3E%3C/rect%3E%3C/svg%3E");
+			--bar-text-color: var(--wp--preset--color--black);
+			--bar-link-color: var(--wp--preset--color--black);
+		}
+
+		.category-events & {
+			--bar-background-image: url("assets/svg/brush-stroke-nav-bar-white.svg"),
+			url("data:image/svg+xml,%3Csvg height='1000' width='1' version='1.1' xmlns='http://www.w3.org/2000/svg'%3E%3Crect x='0' y='0' height='1000' width='1' r='0' rx='0' ry='0' fill='%23ffffff'%3E%3C/rect%3E%3C/svg%3E");
+			--bar-text-color: var(--wp--preset--color--black);
+			--bar-link-color: var(--wp--preset--color--blue-1);
+		}
+
+		.category-month-in-wordpress & {
+			--bar-background-image: url("assets/svg/brush-stroke-nav-bar-dark-grey.svg"),
+			url("data:image/svg+xml,%3Csvg height='1000' width='1' version='1.1' xmlns='http://www.w3.org/2000/svg'%3E%3Crect x='0' y='0' height='1000' width='1' r='0' rx='0' ry='0' fill='%2323282d'%3E%3C/rect%3E%3C/svg%3E");
+			--bar-text-color: var(--wp--preset--color--white);
+			--bar-link-color: var(--wp--preset--color--white);
+		}
+
+		.category-security & {
+			--bar-background-image: url("assets/svg/brush-stroke-nav-bar-coral-red.svg"),
+			url("data:image/svg+xml,%3Csvg height='1000' width='1' version='1.1' xmlns='http://www.w3.org/2000/svg'%3E%3Crect x='0' y='0' height='1000' width='1' r='0' rx='0' ry='0' fill='%23f86368'%3E%3C/rect%3E%3C/svg%3E");
+			--bar-text-color: var(--wp--preset--color--black);
+			--bar-link-color: var(--wp--preset--color--black);
+		}
+	}
+
+	&-breadcrumb {
+		color: var(--bar-text-color);
+
+		@include break-mobile {
+			line-height: 50px;
+		}
+
+		&-current {
+			display: inline-block;
+			font-family: var(--wp--preset--font-family--inter);
+			font-size: 1rem;
+			font-weight: 700;
+
+			[class*="wp-container-"] & {
+				margin-top: 0;
+			}
+
+			&:before {
+				content: "\0020\00B7\0020";
+			}
+		}
+
+		.wp-block-navigation {
+			display: inline-block;
+		}
+
+		a {
+			color: var(--bar-link-color);
+		}
+	}
+
+	&-categories {
+		--flex-items-per-column: 8;
+
+		@include break-medium {
+			--flex-items-per-column: 4;
+		}
+
+		[class*="wp-container-"] & {
+			margin-top: 0;
+		}
+
+		// This selector overcomes the overly generic `input[type=checkbox] + label` rule in the reset.
+		label#{&}-label {
+			display: block;
+			color: var(--bar-text-color);
+			margin: 0;
+			padding: 15px 0;
+			cursor: pointer;
+
+			@include break-mobile {
+				line-height: 50px;
+				padding: 0;
+				text-align: right;
+			}
+
+			span {
+				padding: 0 3px 0 0;
+			}
+
+			svg {
+				vertical-align: middle;
+			}
+		}
+
+		&-toggle {
```

Comment:
The partial class names feels kind of awkward to me; if I don't remember what the containing SCSS block is named, then I have to scroll back up to the top to find out.

PhpStorm does show it in its breadcrumb toolbar, but it still feels less readable to me, especially in long blocks like this.

I don't feel strongly though.